File: 2044 Count Number of Maximum Bitwise-OR Subsets/2044.cpp
class Solution {
public:
    void find(vector<int>& nums, int i, int val, int& maxOr, int& ans){
        if (i == nums.size()) {
            if (val == maxOr) ans++;
        } else {
            find(nums, i+1, val, maxOr, ans);
            find(nums, i+1, val | nums[i], maxOr, ans); 
        }
    }

    int countMaxOrSubsets(vector<int>& nums) {
        int maxOr = 0;

        for (int num: nums) maxOr |= num;

        int ans = 0;

        find(nums, 0, 0, maxOr, ans);

        return ans;
    }
};

File: 2044 Count Number of Maximum Bitwise-OR Subsets/2044.py
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        max_or = 0
        for num in nums:
            max_or |= num
        
        self.ans = 0

        def find(i, val):
            if i == len(nums):
                if val == max_or:
                    self.ans += 1
            else:
                find(i + 1, val)
                find(i + 1, val | nums[i])

        find(0, 0)

        return self.ans
